janusgraph-server/src/main/java/org/janusgraph/graphdb/tinkerpop/gremlin/server/auth/JanusGraphAbstractAuthenticator.java

```diff
@@ -114,8 +115,33 @@ public void setup(final Map<String,Object> config) {
         }
 
         final String defaultUser = config.get(CONFIG_DEFAULT_USER).toString();
-        if (credentialStore.findUser(defaultUser) == null) {
-            credentialStore.createUser(defaultUser, config.get(CONFIG_DEFAULT_PASSWORD).toString());
+        if (!userExists(defaultUser)) {
+            createUser(defaultUser, config.get(CONFIG_DEFAULT_PASSWORD).toString());
+        }
+    }
+
+    protected Vertex findUser(final String username) {
+        credentials.tx().rollback();
```

Comment:
Why do you call rollback?

Reply:
I basically just copied these methods over from TinkerPop's `CredentialGraph` and didn't want to change anything about their functionality. `CredentialGraph.findUser` [looks like this](https://github.com/apache/tinkerpop/blob/ea1e5a323a57d265bc3e588c6e40a30dfc2e8c42/gremlin-groovy/src/main/java/org/apache/tinkerpop/gremlin/groovy/jsr223/dsl/credential/CredentialGraph.java#L56):

```java
public Vertex findUser(final String username) {
        if (supportsTransactions) g.tx().rollback();
        final GraphTraversal<Vertex,Vertex> t = g.V().has(CredentialGraphTokens.PROPERTY_USERNAME, username);
        final Vertex v = t.hasNext() ? t.next() : null;
        if (t.hasNext()) throw new IllegalStateException(String.format("Multiple users with username %s", username));
        return v;
}
```

Since JanusGraph supports transactions, the first line does the rollback.

I assume that this rollback is done to prevent a situation where credentials that were not committed at all are used to authenticate a user. That could be very confusing when the authentication doesn't work any more later one if these credentials were never committed.

Reply:
Make sense.